pkgs/tools/typesetting/tex/texlive/default.nix

```diff
@@ -144,6 +304,59 @@ let
     xzcat "$tlpdb" | sed -rn -f "$tl2nix" | uniq > "$out"
   '';
 
+  # build tlType == "bin" containers based on `binfiles` in TLPDB
+  # see UPGRADING.md for how to keep the list of shebangs up to date
+  # TODO: add extraBuildInputs (?) for adding PATH entries to texlive.combine wrappers
+  mkPkgBin = pname: version: run: { binfiles, scriptsFolder ? pname, postFixup ? "", formats ? [], ... }@args:
+    runCommand "texlive-${pname}.bin-${version}"
+      ( {
+          # metadata for texlive.combine
+          passthru = {
+            inherit pname version;
+            tlType = "bin";
+          };
+          # shebang interpreters
+          # TODO: package SNOBOL4
+          buildInputs = [ bash perl python3 bin.luatex ruby tk ];
+          run = run.outPath or "";
+          inherit scriptsFolder;
+        }
+      )
+      # look for scripts
+      # the explicit list of extensions avoid non-scripts such as $binname.cmd, $binname.jar, $binname.pm
+      # the order is relevant: $binname.sh is preferred to other $binname.*
+      (''
+        mkdir -p "$out/bin"
+        for binname in "${builtins.concatStringsSep "\" \"" binfiles}" ; do
+          target="$out/bin/$binname"
+
+          for bin in '${bin.core}'/bin/"$binname" ${lib.optionalString (bin ? ${pname}) ("'" + bin.${pname}.outPath + "'/bin/\"$binname\"")}; do
+            if [[ -e "$bin" && -x "$bin" ]] ; then
+              ln -s "$bin" "$target"
+              continue 2
+            fi
+          done
+
+          if [[ -n "$run" ]] ; then
+            for script in "$run/scripts/$scriptsFolder/$binname"{,.sh,.lua,.pl,.py,.rb,.sno,.tcl,.texlua,.tlu}; do
+              if [[ -e "$script" ]] ; then
+                sed -e '1s|^eval '"'"'(exit \$?0)'"'"' && eval '"'"'exec perl -S .*$|#!${perl}/bin/perl|' \
+                    -e 's|exec java |exec ${jdk11}/bin/java |g' \
```

Comment:
is there a particular reason for using for using `jdk11` and not `jdk` (or the latest lts version, 17)?

Reply:
No, just me not quite knowing how to deal with Java in Nixpkgs, I'll switch to `jdk`.